inertia/middleware.go

```diff
@@ -0,0 +1,47 @@
+package inertia
+
+import (
+	"context"
+	"net/http"
+)
+
+type MiddlewareFunc func(http.Handler) http.Handler
+
+func Middleware(inertia *Inertia) MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := context.WithValue(r.Context(), inertiaCtxKey, inertia)
+			req := r.WithContext(ctx)
+
+			if r.Header.Get("X-Inertia") == "" {
+				next.ServeHTTP(w, req)
+				return
+			}
+
+			if r.Method == "GET" && r.Header.Get("X-Inertia-Version") != inertia.GetVersion() {
+				w.Header().Add("X-Inertia-Location", r.URL.String())
+				w.WriteHeader(http.StatusConflict)
```

Comment:
Is the idea to return early instead of in the actual handler? Add a comment

Reply:
Yes, inertia will catch the 409 response client side and make a new hard visit to fetch the latest version assets.